ui/src/app/applications/components/applications-list/applications-list.tsx

```diff
@@ -150,14 +150,26 @@ const ViewPref = ({children}: {children: (pref: AppsListPreferences & {page: num
 function filterApps(applications: models.Application[], pref: AppsListPreferences, search: string): {filteredApps: models.Application[]; filterResults: FilteredApp[]} {
     applications = applications.map(app => {
         let isAppOfAppsPattern = false;
+        const childApps = [];
         for (const resource of app.status.resources) {
             if (resource.kind === 'Application') {
                 isAppOfAppsPattern = true;
-                break;
+                childApps.push(resource.name);
             }
         }
-        return {...app, isAppOfAppsPattern};
+        return {...app, isAppOfAppsPattern, childApps};
     });
+    const parentAppArray = applications.filter(app => app.isAppOfAppsPattern);
```

Comment:
Rather than using `filter` here, it would be more efficient to push parent apps to `parentAppArray` in the previous map. E.g.: 

```tsx
const parentAppArray = [];
applications = applications.map(app => {
    ...
    for (const resource of app.status.resources {
        ...
    }
    if (isAppOfAppsPattern) {
        parentAppArray.push(app);
    }
}

Reply:
Hi @rbreeze I tested your solution and found that it would miss `childApss` property in the object of `parentAppArray`. So I had to keep it old way. Thanks for the suggestion though, I thought it would work too.